odl/discr/discr_mappings.py

```diff
@@ -467,219 +482,307 @@ def _call(self, x, out=None):
+            neighbor.
+        order : {'C', 'F'}, optional
+            Ordering of the values in the flat data arrays. 'C'
+            means the first grid axis varies slowest, the last fastest,
+            'F' vice versa.
+        """
+        if not isinstance(ip_fspace, FunctionSpace):
+            raise TypeError('function space {!r} is not a `FunctionSpace` '
+                            'instance.'.format(ip_fspace))
+        if not isinstance(ip_fspace.domain, IntervalProd):
+            raise TypeError('function space domain {!r} is not an '
+                            '`IntervalProd` instance.'.format(ip_fspace))
+
+        FunctionSetMapping.__init__(self, 'extension', ip_fspace, grid, dspace,
+                                    order, linear=True)
+
+        try:
+            schemes_ = str(schemes + '').lower()  # pythonic string check
+            schemes_ = [schemes_] * self.grid.ndim
+        except TypeError:
+            schemes_ = [str(scm).lower() for scm in schemes]
+
+        if nn_variants is None:
+            variants_ = ['left'] * self.grid.ndim
+        else:
+            try:
+                variants_ = str(nn_variants + '').lower()  # pythonic str check
+                variants_ = [variants_] * self.grid.ndim
+            except TypeError:
+                variants_ = [str(var).lower() for var in nn_variants]
+
+        for i, (scm, var) in enumerate(zip(schemes, variants_)):
+            if scm not in SUPPORTED_INTERP_SCHEMES:
+                raise ValueError("Interpolation scheme '{}' at index {} not "
+                                 "understood.".format(scm, i))
+            if scm == 'nearest' and var not in ('left', 'right'):
+                raise ValueError("Nearest neighbor variant '{}' at index {} "
+                                 "not understood.".format(var, i))
+
+        self._schemes = schemes_
+        self._nn_variants = variants_
+
+    @property
+    def schemes(self):
+        """List of interpolation schemes, one for each axis."""
+        return self._schemes
+
+    @property
+    def nn_variants(self):
+        """List of nearest neighbor variants, one for each axis."""
+        return self._nn_variants
+
+    def _call(self, x, out=None):
+        """Create an interpolator from grid values ``x``.
+
+        Parameters
+        ----------
+        x : `FnBaseVector`
+            The array of values to be interpolated
+        out : `FunctionSpaceVector`, optional
+            Vector in which to store the interpolator
+
+        Returns
+        -------
+        out : `FunctionSpaceVector`
+            Per-axis interpolator for the grid of this operator. If
+            ``out`` was provided, the returned object is a reference
+            to it.
+        """
+        def interpolator(arg, out=None):
+            """Interpolating function with vectorization."""
+            if is_valid_input_meshgrid(arg, self.grid.ndim):
```

Comment:
Hm, the code must have been more different before. Can be easily unified, right.